client-tools/Utils/ClientTaskQueue.h

```diff
@@ -394,18 +404,27 @@ inline bool ClientTaskQueue<JobData>::Worker::isIdle() const noexcept {
 template<typename JobData>
 inline void ClientTaskQueue<JobData>::Worker::run() {
   while (!isStopping()) {
+    // we must unconditionally report as busy here, regardless of if there is
+    // actually a job on the queue or not. this is because otherwise there can
+    // be a race between
+    // - this thread popping a job from the queue and only after that reporting
+    // busy.
+    // - someone checking if there are no jobs left and all workers being idle.
+    //
+    _idle.store(false);
```

Comment:
What about set idle under queue mutex?

Reply:
There is currently no queue mutex.
There are 2 condition variables (with associated mutexes) for _jobs and _workers, but they are located on a different object (the queue) whereas the idle flag is located on a Worker thread object.

Anyway, I changed things around so that:
- we only have one condition variable (including its mutex) for _jobs, _workers and the _queue. Previously we had 2 different ones, which allowed for all kinds of races. With the single cv everything is a lot easier. The downside is that we may have more contention than before, but that shouldn't be a problem. We are using this code only for dump/restore, and the queue management cost should not matter much here
- the atomic `_idle` variable is now set and checked under that condition as well.